drud/ddev
# Remove obsolete ddev plugins

## The Problem:

We are abandoning all plugins except the plugin formerly known as 'local' at this time. This removes them.

## The Fix:

Remove them, change names, remove newmedia, temporarily disengage tests by renaming the files.


## Related Issue Link(s):

[Remove legacy and drudapi plugins](https://github.com/drud/ddev/issues/2)

## Release/Deployment notes:
Does this affect anything else, or are there ramifications for other code? Does anything have to be done on deployment?

Several TODOs are in the code - 
* Add tests back in when the local plugin is stable. This is critical because most tests are now removed.
* Fix the environment specifiers when we remove environment config "ActiveApp" and "ActiveDeploy"
* Temporary config hack doesn't read the drud.yaml, and we don't expect to - all it does is default to the values given on the command line, and uses a wired-in workspace dir (~/.drud)

## Code Review

**This is hard to review so I split it into 3 commits.**
* 4faf323 removes a whole pile of obsolete files. It's easier if you don't look at that, maybe just review the filenames that are being reviewed. 
* 8e5b845 has all the logic changes. Most of your attention should be there. 
* b9d6252 mostly just puts tests out of the way temporarily

## Manual Testing

The "-p local" argument is no longer necessary with this PR. That's the only platform plugin remaining and it's the default.

The split of "add" into "start" and "import" is *not* in this PR, so this uses the traditional "add" subcommand.

1. `ddev add <sitename> <environment>` (fails, but creates dirs in ~/.drud/local). Example: `ddev add drud-d8 production`
2. Copy newmedia-style archive tarball into ~/.drud/local but rename it to `<environment>-<sitename>.tar.gz`
3. Untar newmedia-style archive tarball into ~/.drud/local/<sitename>-<environmentname>/src
4.  Rerun `ddev add <sitename> <environment`, Example: `ddev add drud-d8 production` (should work this time)
5. Visit the site, optionally log in, etc.
6. Try the other commands on the site: list, start, stop, restart, ssh, version
